google-marlin: 357 automatically checked security related patches

SkewedZeppelin opened the pull request:
Untested, likely to work.
Unsure if patches block can be replaced with wildcard.
Also not all patches are true patches, some are plain diffs.

An example of #383

samueldr commented:
Converted to draft to improve my ability to deal with the PRs.

Note that I'm still somewhat interested in making this work, but only *somewhat*.

The main problem is that I don't really think working from vendor kernels is a good option for headache-free maintaining. So they have been de-prioritized a lot with regard to development time.

But still, I think it would be worthwhile to package the updater thing such that it can be ran trivially for our needs here.

SkewedZeppelin commented:
fwiw I've updated this, the database had many additions and also replaced many plain diffs with the actual patches, plus I added the concept of loose versioning which will apply 4.4 patches to 3.18 with a selection of known disabled patches

https://github.com/SkewedZeppelin/mobile-nixos/commit/96124593366dacba66661783dfae98d6a2fe7713

samueldr commented:
Yes, I looked at the status upstream and saw there were many updates since. And I fully expect work would be needed.

> completely fair, and awesome to see projects like NixOS and postmarketOS mainlining devices
> just would like to see people using these other devices not have completely insecure kernels ;)

Yes, if time allows, I want to explore this approach. Even if mainline kernels are preferred (for obvious reasons), making use of vendor kernels in a safer manner helps democratizing access for tinkerers to a non-Android Mobile Linux ecosystem.

ahoneybun commented:
Not meaning to bring a possible stale Draft back but could this also effect the Pixel itself (non-XL)?

samueldr commented:
Hi! :wave: 

I have taken the (not so hard) decision to drop support for vendor-based devices.

I'm closing this PR since it only targets working with vendor-based devices. It is not a judgement of the quality of the changes, but really that I don't think it's healthy to keep hoping they may end-up working fine in that state.

This PR would have been useful if we wanted to continue supporting that dead-end. Though it's quite obvious supporting mainline only is both easier and better.

(See also: #643)
